### ISSUE TITLE
Dropout issue on DBN

### ISSUE THREAD

**Issue by mrdeanplumbley** [eclipse/deeplearning4j#91]
Hi, 

I am getting an error when I add dropout in the conf section of a DBN network when I do:
"d.getLayers()[0].conf().setDropOut(0.3f);"
or 
 NeuralNetConfiguration conf = new NeuralNetConfiguration.Builder()
..dropOut(0.3f)

Exception in thread "main" java.lang.IllegalArgumentException: Illegal index 1 derived from 1 with offset of 0 and stride of 1
    at org.nd4j.linalg.api.ndarray.BaseNDArray.linearIndex(BaseNDArray.java:2655)
    at org.nd4j.linalg.api.ndarray.BaseNDArray.getDouble(BaseNDArray.java:2977)
    at org.nd4j.linalg.api.ndarray.BaseNDArray.gti(BaseNDArray.java:799)
    at org.nd4j.linalg.api.ndarray.BaseNDArray.gti(BaseNDArray.java:742)
    at org.nd4j.linalg.api.ndarray.BaseNDArray.gt(BaseNDArray.java:737)
    at org.deeplearning4j.nn.BaseNeuralNetwork.applyDropOutIfNecessary(BaseNeuralNetwork.java:639)
    at org.deeplearning4j.models.featuredetectors.rbm.RBM.sampleHiddenGivenVisible(RBM.java:337)
    at org.deeplearning4j.models.featuredetectors.rbm.RBM.getGradient(RBM.java:163)
    at org.deeplearning4j.optimize.optimizers.NeuralNetworkOptimizer.getValueGradient(NeuralNetworkOptimizer.java:169)
    at org.deeplearning4j.optimize.optimizers.rbm.RBMOptimizer.getValueGradient(RBMOptimizer.java:49)
    at org.deeplearning4j.optimize.solvers.VectorizedNonZeroStoppingConjugateGradient.optimize(VectorizedNonZeroStoppingConjugateGradient.java:134)
    at org.deeplearning4j.optimize.optimizers.NeuralNetworkOptimizer.train(NeuralNetworkOptimizer.java:99)
    at org.deeplearning4j.models.featuredetectors.rbm.RBM.fit(RBM.java:523)
    at org.deeplearning4j.models.classifiers.dbn.DBN.pretrain(DBN.java:201)
    at org.deeplearning4j.models.classifiers.dbn.DBN.pretrain(DBN.java:73)
    at org.deeplearning4j.nn.BaseMultiLayerNetwork.fit(BaseMultiLayerNetwork.java:1359)
    at org.deeplearning4j.nn.BaseMultiLayerNetwork.fit(BaseMultiLayerNetwork.java:1370)

Not 100% sure what I am doing wrong. 

Thanks for the help

Dean
